### PR TITLE
chore(provider): PagerDuty scopes added and fix: mongodb connection check

### LINKED ISSUE #1031: `validate_scopes` for PagerDuty
The DoD if to have `validate_scopes` implemented for `PagerdutyProvider`.
Inspiration can be driven from any provider that implements `validate_scopes` (e.g., Datadog, Grafana, Appdynamics, Dynatrace, etc.).

14-days free trial can be created in: https://www.pagerduty.com/sign-up/

### PR DESCRIPTION
**Pull request by KanvaBhatia**
Closes #1031 and #903

## 📑 Description

Added `validate_scopes` for `PagerdutyProvider`. 
Included the following scopes - 
1. `incidents.read`
2. `incidents.write`
3. `webhook_subscriptions.read`
4. `webhook_subscriptions.write`

- [x] Scopes added and created `validate_scopes`
- [x] Edit Documentation - Add Scopes, Edit Inputs and Authentication Parameters

Also fixed mongodb always validating scopes.

## ✅ Checks

- [x] My pull request adheres to the code style of this project
- [x] My code requires changes to the documentation
- [x] I have updated the documentation as required
- [x] All the tests have passed

## ℹ Additional Information

Pagerduty - 
If api key is wrong (Unauthorized) - 
image

If valid api key and authorized - 
image

Mongodb - 
If wrong URI - 
image

If right URI -
image

### CODE REVIEW

File: keep/providers/pagerduty_provider/pagerduty_provider.py [hunk 1]
```diff
@@ -71,6 +99,37 @@ def validate_config(self):
...
+                elif scope.name.startswith("webhook_subscriptions"):
+                    response = requests.get(
+                        self.SUBSCRIPTION_API_URL,
+                        headers=headers,
+                    )
+                if response.ok:
+                    scopes[scope.name] = True
+                else:
+                    scopes[scope.name] = response.reason
+            except Exception as e:
```
Comment (talboren) [inline keep/providers/pagerduty_provider/pagerduty_provider.py:129]:
```suggestion
            except Exception:
```

File: keep/providers/pagerduty_provider/pagerduty_provider.py [hunk 2]
```diff
@@ -71,6 +99,37 @@ def validate_config(self):
...
+                self.logger.error("Error validating scopes", extra={"error": str(e)})
```
Comment (talboren) [inline keep/providers/pagerduty_provider/pagerduty_provider.py:130]:
```suggestion
                self.logger.exception("Error validating scopes")
```

**Review by talboren — CHANGES_REQUESTED** [review]
Amazing, thanks! 👑 two small comments and linter yelling about some unused variable and we’re set!

**Comment by KanvaBhatia** [comment]
Hey @talboren I by mistake pushed the mongodb changes also in the same branch and they became part of the same PR.
Added the details of that fix in the initial comment.
Fixed the unused variable `e` in mongodb provider and also `logger.exception` instead of `logger.error`.

**Comment by talboren** [comment]
> Hey @talboren I by mistake pushed the mongodb changes also in the same branch and they became part of the same PR. Added the details of that fix in the initial comment. Fixed the unused variable `e` in mongodb provider and also `logger.exception` instead of `logger.error`.

No worries! Looks great.

**Review by talboren — APPROVED** [review]
LGTM

**Pull request merged**